[{"client_msg_id":"5fc5e696-4f03-4445-b4d8-469ef7085b79","type":"message","text":"I find the quality and maturity of Pluto to be very low, and I’m annoyed. However I do need to use it because of Computational Thinking. Here are two examples:\n\nSeveral times in the last weeks I’ve had the UI loose connection with the kernel somehow. I may be able to write something on the UI, but nothing gets executed. When I kill Pluto on the REPL and re-start things, I see that latest things I wrote on the notebook are not saved.\n\nSecond is about the with_terminal(). I tried figuring out why my function is not working properly so I went to the trouble of copy-pasting the code that calls it to another cell, edited the cell so it properly calls that. Then I invoke the cell, verify that I can get some output, but because the internal code throws a MethodError, it no longer displays the output when I really need it. Ok, I try to look for the output in the actual terminal, but because the code is called by several cells in the notebook, I have way too much output. Then I edit the function name so the other cells don’t call it.\n\nNone of this would have been a problem with a Jupyter notebook.","user":"U01CQTKB86N","ts":"1615537866.134400","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"L/0","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"I find the quality and maturity of Pluto to be very low, and I’m annoyed. However I do need to use it because of Computational Thinking. Here are two examples:\n\nSeveral times in the last weeks I’ve had the UI loose connection with the kernel somehow. I may be able to write something on the UI, but nothing gets executed. When I kill Pluto on the REPL and re-start things, I see that latest things I wrote on the notebook are not saved.\n\nSecond is about the with_terminal(). I tried figuring out why my function is not working properly so I went to the trouble of copy-pasting the code that calls it to another cell, edited the cell so it properly calls that. Then I invoke the cell, verify that I can get some output, but because the internal code throws a MethodError, it no longer displays the output when I really need it. Ok, I try to look for the output in the actual terminal, but because the code is called by several cells in the notebook, I have way too much output. Then I edit the function name so the other cells don’t call it.\n\nNone of this would have been a problem with a Jupyter notebook."}]}]}],"thread_ts":"1615537866.134400","reply_count":1,"reply_users_count":1,"latest_reply":"1615544189.134500","reply_users":["U01B75VG3GR"],"subscribed":false},{"client_msg_id":"c9580a5a-2a8e-40a1-9a02-84e08d2428dd","type":"message","text":"If you were just going to dunk on Pluto, could you use the <#C67TK21LJ|gripes> channel for that next time so this channel doesn't get cluttered unnecessarily?","user":"U01B75VG3GR","ts":"1615544189.134500","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"wHjfX","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"If you were just going to dunk on Pluto, could you use the "},{"type":"channel","channel_id":"C67TK21LJ"},{"type":"text","text":" channel for that next time so this channel doesn't get cluttered unnecessarily?"}]}]}],"thread_ts":"1615537866.134400","parent_user_id":"U01CQTKB86N"}]